### ISSUE TITLE
Icon colour categorization (Meta issue)

### ISSUE THREAD

**Issue by polarbearing** [gravitystorm/openstreetmap-carto#3395]
There are a lot of new icons being added recently, and often the colour has to be decided. Doing that on a case-by-case basis apparently leads to inconsistencies, thus this issue tries to collect definitions and decisions, which I can refine in the top post according to the discussion evolving. I focus on the principal colours and leave out fine tuning of labels etc. 

### amenity-brown:  = memorials:, culture:, public-service:
734a08; ![#734a08](https://placehold.it/15/734a08/000000?text=+)
financial services (bank, ATM, bureau_de_change, casino)
unattended shelters and installations (amenity_shelter,  bbq)
unattended items (post_box, recycling, telephone, picnic_table, waste_disposal, bench, waste_basket)
artwork, memorials
culture (theatre, cinema, museum, community_centre, arts_centre
public_bath, shower, toilets, drinking_water, firepit
public service (fire station, embassy, courthouse, social_facility, town_hall, post_office, prison
tourism=office (don't see the rendering)
viewpoint
historic things (castle, fort, archaeological site etc)
_problematic:_ car_wash, car_service

### transportation-icon:  = accommodation-icon:
0092da; ![#0092da](https://placehold.it/15/0092da/000000?text=+)
Transport related (public transport, vehicle rent, taxi, filling/charging station, toll, elevator, parking)
Touristic accommodation (hotel, motel, hostel, tourism-apartment, guest house, camping/caravan)
serviced shelters (alpine_hut,
wilderness_hut appears more to be an unserviced shelter? 

### shop-icon: 
ac39ac; ![#ac39ac](https://placehold.it/15/ac39ac/000000?text=+)
_Shop (wiki) A place selling retail products or services._
shops, markets
driving_school
hygiene services (hair dresser, nail studio)

### health-color: 
da0092; ![#da0092](https://placehold.it/15/da0092/000000?text=+)
Hospital, Pharmacy, doctors, vet, clinic

### gastronomy: PR in prep for own colour
_Services where ready-to-eat food and drinks are served_ 
restaurant, fastfood, pub, cafe, bar, 
including those servings in a cultural context (nightclub)

### religious black 
000000 (needs a variable on top of file) ![#000000](https://placehold.it/15/000000/000000?text=+)
Place of worship, man-made=cross, wayside_cross

### power black
"black" (needs a variable on top of file) ![#000000](https://placehold.it/15/000000/000000?text=+)
generator

### water-color:
fountain
waterfall (as water-text)

### office: 
4863A0; ![#4863A0](https://placehold.it/15/4863A0/000000?text=+)
office=*

### man-made-icon: = advertising-grey:
555; ![#555555](https://placehold.it/15/555555/000000?text=+)
mast, various towers, chimney, city_gate, ad column, lighthouse
bunker
tourism=board,guidepost (don't see the rendering)
wayside_shrine (should that be religious?)
windmill
hunting_stand

### leisure-green: 
darken(@park, 60%); i.e. 0C8416 ![#0C8416](https://placehold.it/15/0C8416/000000?text=+)
sport/play-related
water_park, beach_resort, fitness, dog_park, picnic_site
massage shop, 
sauna
outdoor_seating
bird_hide

### airtransport:
8461C4; ![#8461C4](https://placehold.it/15/8461C4/000000?text=+)
helipad, aerodrome
ferry_terminal (why is that on air and not water or transport?)

### landform-color:
d08f55; ![#d08f55](https://placehold.it/15/d08f55/000000?text=+)
natural peak, volcano, saddle

### undefined - marker-fill is missing
natural_cave_entrance (renders black)
mini_roundabout (renders 666666)
railway level crossing, thin line sampled as 4b4c4a probably dithered with the halo
barrier=gate (also, 404040)

### shades of grey
lift_gate, swing_gate 3f3f3f ![#3f3f3f](https://placehold.it/15/3f3f3f/000000?text=+)
bollard, block, log 7d7c7c ![#7d7c7c](https://placehold.it/15/7d7c7c/000000?text=+)

**Comment by polarbearing** [comment]
in discussion: craft  #3126, bike repair station #3069 #3391

**Comment by polarbearing** [comment]
![#f03c15](https://placehold.it/15/f03c15/000000?text=+)

**Comment by jeisenbe** [comment]
Excellent idea for discussion.

One simple improvement would be unifying the airtransport and
transportation-icon colors.
This would keep blue tones for water features only.
@imagico suggested these changes previously on his blog

Drinking water should then be changed to water-color, like waterfalls and
fountains.
I'm not sure if shower and public_bath should be amenity-brown or
water-color

Wilderness_hut is similar to camping / caravan / alpine_hut, so those
should all stay together.
But I don't know if transport/accomodation is better than leisure for these

As far as the color choices go, most are good, but I wonder how #ac39ac was
picked for shops.
If I was starting from scratch, I would have picked a reddish color similar
to landuse=retail (but more prominent)

-Joseph Eisenberg

**Comment by polarbearing** [comment]
Thanks for the comments. I'd like to keep this discussion for the categories, not tuning a particular colour. 

> Wilderness_hut is similar to camping / caravan / alpine_hut

No it is particularly not. [Wilderness hut](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dwilderness_hut) is unstaffed and provides protection from the elements, much like amenity=shelter. An [Alpine Hut](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dalpine_hut) is serviced and provides lodging and meals. 

As for features using water, such as drinking_water, shower and public_bath, I'm not convinced to move them to water_colour, as the service character prevails over the medium. In car_wash the cars get wet as well, and it is more a service (which I'd put into shop colour).

**Comment by jeisenbe** [comment]
Sorry about the confusion, I'll stick to discussing the categories now.

1) Shelters/Campgrounds: I see that amenity
<https://wiki.openstreetmap.org/wiki/Key:amenity>=shelter
<https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dshelter> plus
shelter_type <https://wiki.openstreetmap.org/wiki/Key:shelter_type>=
basic_hut is similar to wilderness_hut, except for the lack of fireplace
(and probably toilets / running water), and these shelters rendered with
amenity-brown, so that is a point in favor of moving wilderness_hut to that
category.

Do camp sites and caravan sites provide serivices? In the northwestern USA,
I believe the majority of campgrounds and RV sites have pretty limited
services (though some have full services), and camping sites in the
backcountry are very basic. The wiki says "There are two types of
campsites:  a dedicated area with improvements and various facilities an
impromptu area (as one might decide to stop while backpacking or hiking)".
Should campgrounds be categorized as transport/accomodation?

2) Water amenities: I don't mind if showers and baths are in amenity-brown;
these are amenities that happen to use water, just like a car wash (as you
mentioned).
But I think drinking water is a water feature; access to water is the
"amenity". And there is a plan to add man_made=water_well and water_pump.
These should be the same color ad drinking water or water_tap or natural
springs: (https://github.com/gravitystorm/openstreetmap-carto/issues/1224)

3) Transport: I still think transport should be unified with airtransport,
no matter what color is picked.

**Comment by Tomasz-W** [comment]
@polarbearing Thanks for complex analysis of colours thing. I think it shows good why we need to recolour gastronomy key - as most of colours are predicted for 1-2 groups, amenity-brown is predicted for 4 groups of tags (!). It's surely too much. I hope next step will be https://github.com/gravitystorm/openstreetmap-carto/issues/3326 , as tags included in this issue are not memorials, not cultural-like and not a service.

**Comment by jeisenbe** [comment]
To consolidate the discussion here, on issue [3326](https://github.com/gravitystorm/openstreetmap-carto/issues/3326), @Tomasz-W recommended changing unattended shelters and installations (shelter, bbq, bench, picnic_table, waste_basket, waste_disposal, recycling) to man-made-icon gray, because they are not cultural or public service related. 

I see that outdoor_seating is under leisure-green. However, it is specifically used for outdoor seating at cafes, bars and restaurants: "A seating area, usually for the consumption of food and drink from neighbouring cafes and restaurants, often belonging to one or more of them, but not necessarily adjacent." [wiki](https://wiki.openstreetmap.org/wiki/Tag:leisure=outdoor_seating). Examples are beer gardens, restaurant terraces, food courts with shared seating.
So outdoor_seating should be gastronomy-color.

**Comment by Tomasz-W** [comment]
> So outdoor_seating should be gastronomy-color.

It shouldn't. Outdoor seatings are parts of gastronomy points, not a gastronomy points itself. If we keep this logic, we should render trash can in front or a restaurant in _garstronomy-orange_.

**Comment by dieterdreist** [comment]
sent from a phone

> On 15. Sep 2018, at 08:01, Tomasz Wójcik <notifications@github.com> wrote:
> 
> It shouldn't. Outdoor seatings are parts of gastronomy points, not a gastronomy points itself. If we keep this logic, we should render trash can in front or a restaurant in garstronomy-orange.


+1


wrt to the proposal to group springs and drinking water fountains by using the same color: I would not do it, while it is one way to see it (all water) there are also different points of view possible: natural features vs. amenities (man made), e.g. drinking water, post box, phone booth, ...

**Comment by polarbearing** [comment]
Started to update the top post.

**Comment by polarbearing** [comment]
That does not take any load from the amenity-brown and would go in the opposite direction. 
So what speaks against office-blue for those workplaces, offering paid services and crafts?

**Comment by jeisenbe** [comment]
Retail services are quite different than offices. It's sort of the
difference between retail and commercial landuse: lots of people work in
offices, but most of them have no face-to-face interaction with customers,
unlike at a retail business. Barber shops are tiny but have high
foot-traffic.

The craft places would fit better in the same category as offices, since
the location on the map is often just the head office or the place that
goods are made, and not a retail location that good or services are
delivered to the public.

But we'll need to reconsider the dark blue color if it's going to be used
for bigger icons, not just a small dot, especially if Transportation stays
blue (instead of being merged with Airtransport purple)

**Comment by polarbearing** [comment]
We just assigned the gastronomy as a face-to-face interaction business their own colour, so I see no reason why other f2f services need to remain within amenity-brown.

**Comment by meased** [comment]
Perhaps religious black, power black, man-made grey, and shades of grey should be merged into some sort of a "generic" category. These colors all look the same anyway.

**Comment by polarbearing** [comment]
The categories should be driven semantically, not just grouped by having had a similar colour at some time. Religious is quite specific in my opinion and not generic. Grey and black look different to me.

**Comment by meased** [comment]
> Religious is quite specific in my opinion and not generic. Grey and black look different to me.

Perhaps the categories of grey and black should be defined? Regardless of intention, black seems to be used as a catch-all.

**Comment by polarbearing** [comment]
> Perhaps the categories of grey and black should be defined

Yes indeed. One outcome of the analysis was that there are a few hardcoded colours that should be grouped.

**Comment by jragusa** [comment]
water-color should be unified: `natural=waterfall` uses water-text for both colour and icon whereas `amenity=fountain` uses marina-text.

**Comment by Tomasz-W** [comment]
> That does not take any load from the amenity-brown and would go in the opposite direction.
> So what speaks against office-blue for those workplaces, offering paid services and crafts?

@polarbearing The thing is not to throw out of _amenity-brown_ as much tags as possible, but to make a logical complex system. If we render public-services in _amenity-brown_, it seems to be logical step to extend it also for non-public services (after deloading it by https://github.com/gravitystorm/openstreetmap-carto/pull/3360 and https://github.com/gravitystorm/openstreetmap-carto/issues/3326). When I see _office-blue_, I predict some place full of papers/ suits etc., and as I can understand your an argument that `craft=*` places are sometimes offices, they are rather an exeptions (please see: https://wiki.openstreetmap.org/wiki/Key:craft#Craft). That's the reason why I find _office-blue_ very confusing for `craft=*` key.

**Comment by dieterdreist** [comment]
sent from a phone

> On 17. Sep 2018, at 23:09, Jérémy Ragusa <notifications@github.com> wrote:
> 
> water-color should be unified: natural=waterfall uses water-text for both colour and icon whereas amenity=fountain uses marina-text.
> 


also water treatment and toilets, storage tanks that store water, water towers, weirs, factories that bottle water (product=water), boat rentals? Umbrella merchants?

Can you explain why you believe that having something to do with water is a good criterion for a color category (for using the same color), as opposed to how things are currently grouped?

**Comment by jeisenbe** [comment]
Fountains and Waterfalls are both blue now, but the shade used is
different. I have trouble imagining fountains and waterfalls in any other
category. I’d call the category “water features”

Do you think either of these should be in a different category?

**Comment by polarbearing** [comment]
@jragusa - the waterfall is natural, while the fountain is man-made like the marina. 
Fountain even uses two colours, water for the basin and marina for the nozzles.

@Tomasz-W - I don't get why office-blue reminds you to paper and white-collar workers. Just because it was initially introduced for the office dots? It could easily be renamed in workplace-blue. 

The general problem with the categorisation here is, that for any two objects A and B in the world you may find a criterion that binds them into one category AB. Then you add object C and find that you could create one category AC and one BC, so you end up with as many categories as objects. Adding more objects, your number of potential categories grows rapidly as (n*n-n)/2.

In consequence, it is good that we brainstorm about potential category definition, that's why I opened this thread, but we also need to consolidate them.

**Comment by imagico** [comment]
> Fountains and Waterfalls are both blue now, but the shade used is different. I have trouble imagining fountains and waterfalls in any other category.

For reference:  There are currently three water related POI colors in this style: Fountains, Waterfalls and Springs.  See https://github.com/gravitystorm/openstreetmap-carto/pull/3360#issuecomment-416323313

**Comment by Adamant36** [comment]
Any chances of making all emergency/health services like hospitals, fire departments, police etc their own category in red? It might help offload amenity brown a little more and help them to be more cohesive with each other on the map.

**Comment by jeisenbe** [comment]
Hospitals, clinics, pharmacies etc are health-color: da0092;
Are you suggesting adding fire / police to the same category, or making a
new separate category?

**Comment by Adamant36** [comment]
@jeisenbe, changing the color of healthcare to red is currently being discussed in issue #2938, which looks like will happen. As it looks a lot better and seems to have widespread support there. So I'm suggesting fire/police be added to that category once it gets changed.

**Comment by kocio-pl** [comment]
Maybe such category should be named "emergency"? I'm not sure if emergency phone would belong there, as it fits it, but is small and maybe not that visible (see #2993).

**Comment by jeisenbe** [comment]
I’m not sure that pharmacies, doctors, and clinics fit in an “emergency”
category. Veterinarians, dentists probably don’t.

**Comment by Adamant36** [comment]
Depends on why your going to them. People take their pets to veterinarians for life threatening things all the time like snake bites, ingesting poison, getting hit by cars etc. When I was a kid I went to my normal dentist due to an accident that involved the front of my face impacting a sidewalk a lot harder then it probably should have. Id call that an emergency also. Pharmacies often dispense life saving insulin medication and antibiotics. I know in a lot of healthcare related instances the patents might see things as emergencies that aren't though. At least the color red is associated with those things for the average person and mapper. There will always be exceptions though and I agree the average family doctor probably isn't dealing with emergency situations on a regular basis. Which is why id see it more as an emergency/healthcare category then straight emergency. Calling it straight "emergency" also makes a lot more sense then emergency_healthcare does when it comes to coding.

 I also think Kocio-pl's suggestion of adding emergency phones to it is a good. Although, we could nitpic that they are emergency phones in name only and shouldn't be included, because the real world usage of them is usually more like "My car needs a tow" then "Oh my god, my car just exploded and my arm flew into a field. Send an ambulance." But such is life. At least it frees up some brown and makes more sense thematically then "a fire station and a water fountain both brown because they are public services." I actually didn't even know those were emergency phones until kocio-pl just pointed it out.

**Comment by kocio-pl** [comment]
>  I actually didn't even know those were emergency phones until kocio-pl just pointed it out. 

So maybe we could just turn them into red telephone icon (but maybe mirrored to make it easier to see the difference), so they would be less cryptic and less solid at the same time. Sounds promising for me.

**Comment by jeisenbe** [comment]
Good ideas, everyone.

Ok, Police and Fire are grouped as “Public Safety” at least in the USA. How
about we call the new category of healthcare & public safety something like
“Health & Safety”?

(I agree that red is a good color for these categories)

**Comment by polarbearing** [comment]
Hospital and Fire station differ in the emergency case as for the former the location matters for the user, in the latter not.

**Comment by dktue** [comment]
I definitely like the idea of an emergency category

**Comment by Adamant36** [comment]
Health & Safety sounds good to me. 

@polarbearing, yeah true, but what's your point? I mean, are you saying that's a reason to not render them in the same category or something?

**Comment by kocio-pl** [comment]
We could simply have both categories and use the same color for them. :smile: It will make it easier to adapt if someone wants to.

**Comment by jeisenbe** [comment]
Kocio-pl, that’s a great idea.

Are accommodation and transportation already separate in the code, and just
use the same color? They could both be sub-categories.

Similarly, I’d suggest these subcategories for the icons currently in
Amenity-brown (even if we choose to keep the same color)

1) Financial services:
ATM, Bank, bureau_de_change
(I wouldn’t include Casinos. Maybe they can be leisure or tourism?)

2) Public services:
Fire station, embassy, courthouse, social_facility, town_hall, post_office,
prison, community_centre
(Note that I moved community_centre here from the cultural list)

3) Unattended items: (“street furniture?”)
amenity_shelter, bbq, post_box, recycling, telephone, picnic_table,
waste_disposal, bench, waste_basket, firepit
- should probably not became color as otheramenities

4) Hygiene:
public_bath, shower, toilets, drinking_wate
(Or could also include recycling / waste here)

5) Culture & History:
artwork, memorial, theatre, cinema, museum, arts_centre, gallery, castle,
fort, archeological, plus other historic things

I’m not sure where tourism offices and viewpoints should be included, but
tourism=board and =guidepost are adverting-gray right now.

**Comment by kocio-pl** [comment]
It all started with me trying just to make some sane internal categories for a clean code purposes. :smile: You can look at #2824, #2826, #2827 (leisure icon color introduced), #2854 and #2855.

We can have more categories using the same color, it's even better to have `@culture` and `@history` variables than `@culture_and_history`, also because someone might use them for tuning colors.

**Comment by polarbearing** [comment]
> yeah true, but what's your point?

There's good brainstorming going on here currently. I don't have  a fixed opinion for everything, I also like to listen :-)

**Comment by Adamant36** [comment]
I agree. I just thought there might of been more behind it that I was missing for some reason. 

On a related note, id love to see this issue help lead to the code being cleaner. The amenity file could really use some reorganzing and I wish there was some better documentation in the code about what things do.

**Comment by kocio-pl** [comment]
What exactly would you like to understand?

**Comment by Tomasz-W** [comment]
I'm not sure about emergency-colour idea. We surely need separate colour for healthcare objects, and propably there will be a generic dots for healthcare=* tags in future - I'm worried that putting eg. `healthcare=speech_therapist` (see: https://github.com/gravitystorm/openstreetmap-carto/issues/2938#issuecomment-366990786) in the same category with `amenity=fire_station` or `amenity=police_station` would't be an improvement but rather a confusing mistake. Of course there are many categorisation methods, and these objects can be consider as more or less emergency-like if we want to categorise different aspects of human life, but for map categorisation purposes, I think they are not that close to each other to use the same colour for both of them.

**Comment by kocio-pl** [comment]
I don't think there will be great problems with splitting stylesheets, though I haven't tested it. You just need to include new ones in `project.mml`. It would make them editing stylesheets more clean, but I'm not sure how it would influence the performance - @pnorman, what do you think about it?

**Comment by Adamant36** [comment]
@kocio-pl, that's what I was thinking. Is there anyone else that would know about the possible performance issues besides @pnorman? because I'd like to do it. It would make editing stylesheets easier to edit. Plus, Id like to add documentation at some point and it would help with that also.

**Comment by kocio-pl** [comment]
The problem is that we have no performance framework (see #1287), so no one really knows until production deployment shows if there are such problems or not. There are only some hints when somebody tries development deployment - for example @rrzefox volunteers to do that sometimes, but this is not permanent solution and results depend on the traffic.

**Comment by meased** [comment]
@giggls made this script:

https://github.com/giggls/openstreetmap-carto-de/blob/master/scripts/render_single_tile.py

If we could get it work in the docker environment, we could hand select some tiles that exercise a feature, and then time how long the tiles take to render before and after the change. That should a very good indicator of performance.

**Comment by jeisenbe** [comment]
health-color is now #BF0000; (red), and includes dentists, vets, pharmacies, and many features under healthcare=*

**Comment by jeisenbe** [comment]
railway=tram_stop and railway=station use yet another shade of blue: `@station-color: #7981b0;`

**Comment by jeisenbe** [comment]
While the top post shows `fountain` and `waterfall` as "water-color", these icons currently use unique colors, as does the spring icon. There are 3 water-feature colors currently, and 3 other categories use blue (2 of the transportation-related icon classes, and offices):

1) transportation-icon: `#0092da` - also used for the bus station icon, but this is set in the SVG file.
2) office: `#4863A0` - used for the office dots
3) station-color: `#7981b0` - used for train stations and tram stops
4) marina-text: `#576ddf` - used for the fountain icon
5) water-text: `#4d80b3` - used for the waterfall icon
6) The spring.svg file uses `#7abcec`

I would recommend creating a `@water-icon:` line in amenity-points.mss, and this color could be used for fountains, waterfalls and springs.

I'm also in favor of using the same color for `station-color` and `transportation-icon`, as discussed in #1127

**Comment by jeisenbe** [comment]
barrier=gate seems to be 3f3f3f with a ffffff halo, so same as lift_gate and swing_gate
```
<path d="M 0,0 6,0 6,5 0,5 z" fill="#ffffff" />
<path d="M 0,1 6,1 6,2 0,2 z" fill="#3f3f3f" />
```
Perhaps these and bollard, block, log (`#7d7c7c`) should all be combined with `@man-made-gray`.

**Comment by jragusa** [comment]
@jeisenbe You forgot `#6666ff` for ferry routes. I agree to use a unique colour for all of these water features.

**Comment by kocio-pl** [comment]
I don't feel transport and accommodation should be decoupled. They are quite close to me and I would rather use separate color for craft for example.

**Comment by jeisenbe** [comment]
> #6666ff for ferry routes

That's a linear features, more like roads or railways, so I wouldn't include it in the list of icon colors, but it's a good idea to keep it in mind while considering icon colors.

> I would rather use separate color for craft for example.

I've got 2 new colors in #3641 that we can use, so one of them can be saved for `craft=*` and we can still add a new color for accommodation.

**Comment by jragusa** [comment]
I agree with @kocio-pl I do not see the need to add a specific colour for accommodation but I guess that features with close colours could be simplified with a unique colour. Moreover I don't think that creating a complex colour palette is a good thing because it can disturb people looking at the maps. IMO I prefer creating a more uniform and simple colour scheme because it more easier to people to understand the features on the map.

**Comment by jeisenbe** [comment]
Another addition:  `#d40000` (natureal=volcano icon color)

**Comment by jeisenbe** [comment]
Wind generators have been switched from black to man-made-icon
So the only black icons are natural=cave, places_of_worship icons and crosses.
